include/aspect/compat.h

```diff
@@ -65,6 +65,73 @@ namespace aspect
       }
   };
 }
+
+#include <deal.II/grid/tria_boundary_lib.h>
+
+/**
+ * The SpecialConeBoundary implements the normal_vector function.
+ */
+namespace aspect
+{
+  using namespace dealii;
+
+  template <int dim>
+  class SpecialConeBoundary : public dealii::ConeBoundary<dim>
+  {
+    public:
+
+      /**
+       * Constructor. Here the boundary object is constructed. The points
+       * <tt>x_0</tt> and <tt>x_1</tt> describe the starting and ending points of
+       * the axis of the (truncated) cone. <tt>radius_0</tt> denotes the radius
+       * corresponding to <tt>x_0</tt> and <tt>radius_1</tt> the one corresponding
+       * to <tt>x_1</tt>.
+       */
+      SpecialConeBoundary (const double radius_0,
+                           const double radius_1,
+                           const Point<dim> x_0,
+                           const Point<dim> x_1);
+
+      virtual
+      Tensor<1,dim>
+      normal_vector (const typename Triangulation<dim>::face_iterator &face,
+                     const Point<dim> &p) const;
```

Comment:
Will work on it.

But on a tangent, we only need this class with old deal.II versions anyway, right? In that case, won't the issue go away sometime soon anyway?